static/skin/index.js

```diff
@@ -197,13 +197,63 @@
         }
     }
 
+    /* hack for library.kiwix.org magnet links (created by MirrorBrain)
+        See https://github.com/kiwix/container-images/issues/242 */
+    async function getFixedMirrorbrainMagnet(magnetLink) {
+        // parse as query parameters
+        const params = new URLSearchParams(
+            magnetLink.replaceAll('&amp;', '&').replace(/^magnet:/, ''));
+
+        const zimUrl = params.get('as'); // as= is fallback URL
+        // download metalink to build list of mirrored URLs
+        let mirrorUrls = [];
+        try {
+            const metalink = await fetch(`${zimUrl}.meta4`).then(response => {
+                return response.ok ? response.text() : '';
+            }).catch((_error) => '');
+            if (metalink) {
+                const parser = new DOMParser();
+                const doc = parser.parseFromString(metalink, "application/xml");
+                doc.querySelectorAll("url").forEach((node) => {
+                    if (node.hasAttribute("priority")) { // ensures its a mirror link
+                        mirrorUrls.push(node.innerHTML);
+                    }
+                });
+            }
+        } catch (err) {
+            // not a big deal, magnet will only contain primary URL
+            console.debug(`Failed to retrieve mirror links for ${zimUrl}`);
+        }
```

Comment:
Why is the `try/catch` block needed? If it is a protection against malformed metalink XML then its scope better be narrowed down (and the error message changed accordingly).

Reply:
Exactly, since it's not required per se (magnet will work with a single webseed), I don't want something unexpected to break the magnet link. Could be network-related or parsing related. AFAIK, there's no UI feedback on errors

Reply:
> Could be network-related

Won't network errors be handled by the `catch()` of the `fetch()`?

Reply:
Right. I guess it's only about parsing. Do you think we should drop it?

Reply:
No, the suggestion was to enclose in `try{}` only the code the actually needs it (and adjust the error message).

Reply:
Got it